Sources/Webservice/Webservice.swift

```diff
@@ -33,40 +37,53 @@ public struct Webservice: ResourceRequestable {
          completion: @escaping (Result<Response, Swift.Error>) -> Void) {
 
         do {
-            let requestBehavior = defaultRequestBehavior.and(resource.requestBehaviour)
+            let requestBehavior = defaultRequestBehavior.and(resource.requestBehavior)
 
             let request = requestBehavior.modify(planned: try URLRequest(resource: resource,
                                                                          defaultRequestBehavior: defaultRequestBehavior,
                                                                          baseURL: baseURL))
 
-            requestBehavior.before(sending: request)
+            DispatchQueue.main.async {
```

Comment:
This is technically a dependency and should be injected into the Webservice at somepoint. I think it is also the reason for the tests to fail